pkg/ccl/backupccl/backup_planning.go

```diff
@@ -1131,12 +846,31 @@ func backupPlanHook(
 			CreatedBy: backupStmt.CreatedByInfo,
 		}
 
+		lic := utilccl.CheckEnterpriseEnabled(
+			p.ExecCfg().Settings, p.ExecCfg().ClusterID(), p.ExecCfg().Organization(), "",
+		) != nil
+
+		// We create the job record in the planner's transaction to ensure that
+		// the job record creation happens transactionally.
+		plannerTxn := p.ExtendedEvalContext().Txn
+
+		jobID := p.ExecCfg().JobRegistry.MakeJobID()
+
+		// The protect timestamp logic for a DETACHED BACKUP can be run within the
```

Comment:
we should remove this comment if we are sharing this protection between both detached and non-detached now.